packages/lighthouse-service/src/helpers.ts

```diff
@@ -62,47 +70,70 @@ class LHCIHelper {
-      .then((result: LighthouseProjectType) => result, (error: any) => {
-        throw new Error(error);
-      });
+    })
+      .then((res: any) => res.json())
+      .then(
+        (result: LighthouseProjectType) => result,
+        (error: any) => {
+          throw new Error(error);
+        },
+      );
   }
 
-  fetchProjectBuilds(serverBaseUrl: string, projectID: string, branch:string, limit: number) {
+  fetchProjectBuilds(
+    serverBaseUrl: string,
+    projectID: string,
+    branch: string,
+    limit: number,
+  ) {
     const headers = new Headers();
     headers.append('Content-Type', 'application/json');
-    return fetch(`${serverBaseUrl}/v1/projects/${projectID}/builds?branch=${branch}&limit=${limit}`, {
-      method: 'GET',
-      headers,
-      redirect: 'follow',
-    }).then((res: any) => res.json())
-      .then((result: LighthouseProjectType) => result, (error: any) => {
-        throw new Error(error);
-      });
+    return fetch(
+      `${serverBaseUrl}/v1/projects/${projectID}/builds?branch=${branch}&limit=${limit}`,
+      {
+        method: 'GET',
+        headers,
+        redirect: 'follow',
+      },
+    )
+      .then((res: any) => res.json())
+      .then(
+        (result: LighthouseProjectType) => result,
+        (error: any) => {
+          throw new Error(error);
+        },
+      );
   }
 
   fetchProjectLHR(serverBaseUrl: string, projectID: string, buildID: string) {
     const headers = new Headers();
     headers.append('Content-Type', 'application/json');
-    return fetch(`${serverBaseUrl}/v1/projects/${projectID}/builds/${buildID}/runs`, {
-      method: 'GET',
-      headers,
-      redirect: 'follow',
-    }).then((res: any) => res.json())
-      .then((results: any) => {
-        const scores: LighthouseScoreType[] = [];
-        results.map((value: any) => {
-          const lhr = JSON.parse(value.lhr);
-          const data: any = {};
-          Object.keys(lhr.categories).map((category: any) => {
-            data[camelCase(category)] = lhr.categories[category].score;
+    return fetch(
+      `${serverBaseUrl}/v1/projects/${projectID}/builds/${buildID}/runs`,
+      {
+        method: 'GET',
+        headers,
+        redirect: 'follow',
+      },
+    )
+      .then((res: any) => res.json())
+      .then(
+        (results: any) => {
+          const scores: LighthouseScoreType[] = [];
+          results.forEach((value: any) => {
+            const lhr = JSON.parse(value.lhr);
+            const data: any = {};
+            Object.keys(lhr.categories).forEach((category: any) => {
+              data[camelCase(category)] = lhr.categories[category].score;
+            });
+            scores.push(data);
           });
-          scores.push(data);
-        });
-        return scores;
-      }, (error: any) => {
-        throw new Error(error);
-      });
+          return scores;
+        },
+        (error: any) => {
+          throw new Error(error);
+        },
```

Comment:
You can just put this inside a `.catch()` instead.

Reply:
Done. Also added fetch res.ok check as catch wont work like that in fetch